File: src/main/java/br/com/jiratorio/mapper/transformer/StringTransformer.java
package br.com.jiratorio.mapper.transformer;

import br.com.jiratorio.util.DateUtil;
import java.text.Normalizer;
import java.util.List;
import java.util.stream.Collectors;
import org.mapstruct.Named;
import org.springframework.stereotype.Component;

@Component
public class StringTransformer {

    @Named("toUpperCase")
    public String toUpperCase(final String str) {
        return str == null ? null : str.toUpperCase(DateUtil.LOCALE_BR);
    }

    @Named("listToUpperCase")
    public List<String> listToUpperCase(final List<String> strs) {
        return strs == null ? null : strs.stream().map(String::toUpperCase).collect(Collectors.toList());
    }

    @Named("stripAccents")
    public String stripAccents(final String s) {
        String str = Normalizer.normalize(s, Normalizer.Form.NFD);
        str = str.replaceAll("[\\p{InCombiningDiacriticalMarks}]", "");
        return str;
    }

}

File: src/main/java/br/com/jiratorio/mapper/transformer/StringTransformer.kt
package br.com.jiratorio.mapper.transformer

import org.springframework.stereotype.Component
import java.text.Normalizer

@Component
class StringTransformer {

    fun toUpperCase(str: String?): String? {
        return str?.toUpperCase()
    }

    fun listToUpperCase(strs: MutableList<String>?): MutableList<String>? {
        return strs?.map { it.toUpperCase() }?.toMutableList()
    }

    fun stripAccents(s: String): String {
        var str = Normalizer.normalize(s, Normalizer.Form.NFD)
        str = str.replace("[\\p{InCombiningDiacriticalMarks}]".toRegex(), "")
        return str
    }

}
